js/base/Exchange.js

```diff
@@ -2162,9 +2162,8 @@ module.exports = class Exchange {
             }
             // if it was not defined in market object
             if (subType === undefined) {
-                const exchangeWideValue = this.safeString2 (this.options, 'defaultSubType', 'subType', 'linear');
-                const methodOptions = this.safeValue (this.options, methodName, {});
-                subType = this.safeString2 (methodOptions, 'defaultSubType', 'subType', exchangeWideValue);
+                const values = this.handleOptionAndParams (undefined, methodName, 'subType', 'linear'); // no need to re-test params here
+                subType = values[0];
```

Comment:
`handleOptionAndParams` has the correct flow of logic, so it's better we used it there. 
(additionally, when we don't need to check `params`, we provide undefined there).